mne/preprocessing/tests/test_regress.py

```diff
@@ -136,3 +138,31 @@ def test_read_eog_regression(tmp_path):
     assert_array_equal(model.coef_, model2.coef_)
     assert model.proj == model2.proj
     assert model.info_.keys() == model2.info_.keys()
+
+
+def test_regress_artifact_bads():
+    """Test that bad channels are handled properly."""
+    # Pick the first few EEG channels
+    raw = read_raw_fif(raw_fname).del_proj().set_eeg_reference(projection=True)
+    picks_all = np.concatenate([
+        pick_types(raw.info, meg=True)[:4],
+        pick_types(raw.info, eeg=True)[:8],
+        pick_types(raw.info, eog=True)[:1],
+    ])
+    raw.pick(picks_all).load_data()
+    assert len(raw.ch_names) == 13  # 4 MEG, 8 EEG, 1 EOG
+    del picks_all
+    picks = pick_types(raw.info, eeg=True)
+    assert_array_equal(picks, np.arange(4, 12))
+    norms = np.linalg.norm(raw.get_data(picks), axis=1)
+    raw_reg, _ = regress_artifact(raw, picks=picks, picks_artifact='eog')
```

Comment:
@wmvanvliet this is where in the OPM example it used to fail during the `.apply` stage, saying some of the picks (e.g., 94) were too large for info, which only contained 86 channels following the `pick_info` call